main/src/ca/uwaterloo/flix/language/ast/NamedAst.scala

```diff
@@ -294,6 +303,8 @@ object NamedAst {
 
   case class MatchRule(pat: NamedAst.Pattern, guard: NamedAst.Expression, exp: NamedAst.Expression) extends NamedAst
 
+  case class SelectRule(pat: NamedAst.Pattern, chan: NamedAst.Expression, exp: NamedAst.Expression) extends NamedAst
```

Comment:
We have talked about whether it is a good idea to use a pattern or just a variable here.